aten/src/ATen/native/mps/operations/GridSampler.mm

```diff
@@ -120,6 +131,296 @@ static void grid_sampler_2d_mps_impl(Tensor& output,
     runMPSGraph(stream, cachedGraph->graph(), feeds, outputPlaceholder);
   }
 }
+
+static void grid_sampler_3d_mps_impl(Tensor& output,
+                                     const Tensor& input,
+                                     const Tensor& grid,
+                                     int64_t interpolation_mode,
+                                     int64_t padding_mode,
+                                     bool align_corners) {
+  check_grid_sampler_common(input, grid);
+  check_grid_sampler_3d(input, grid, interpolation_mode);
+
+  auto stream = getCurrentMPSStream();
+  auto device = MPSDevice::getInstance()->device();
+
+  // Kernel selection logic
+  bool use_vectorized = false;
+
+  // Environment variable override
+  const char* env_kernel = std::getenv("PYTORCH_MPS_GRID_SAMPLER_3D_KERNEL");
+  if (env_kernel) {
+    if (std::string(env_kernel) == "vectorized") {
```

Comment:
Nit: use string_view here? No reason to take ownership and allocate on the heap for an equality comparison.